client/src/components/custom/login-dialog/login-dialog.tsx

```diff
@@ -0,0 +1,168 @@
+import { useColorModeValue } from '@/components/ui/color-mode';
+import {
+  DialogBackdrop,
+  DialogBody,
+  DialogCloseTrigger,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogRoot,
+  DialogTitle,
+  DialogTrigger,
+} from '@/components/ui/dialog';
+import { UserCredentials } from '@/types';
+import {
+  closeLoginModalAtom,
+  isLoginModalOpenAtom,
+  openRegisterModalAtom,
+} from '@/utils/atoms/modal-atoms';
+import { useUser } from '@/utils/providers/user-provider/use-user';
+import {
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+} from '@chakra-ui/form-control';
+import { Button, Input } from '@chakra-ui/react';
+import { useAtom } from 'jotai';
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { toast } from 'react-toastify';
+
+const LoginDialog = () => {
+  const [isLoginModalOpen] = useAtom(isLoginModalOpenAtom);
+  const [, closeLoginModal] = useAtom(closeLoginModalAtom);
+  const [, openRegisterModal] = useAtom(openRegisterModalAtom);
+  const [isLoading, setIsLoading] = useState(false);
+  const { login } = useUser();
+
+  const {
+    handleSubmit,
+    register,
+    formState: { errors },
+  } = useForm<UserCredentials>();
+
+  function onSubmit(values: UserCredentials) {
+    setIsLoading(true);
+    login(values)
+      .then(() => {
+        closeLoginModal();
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        console.error(error);
```

Comment:
should we leave this?

Reply:
At this stage of app development, I think we should